code/modules/mob/living/carbon/human/human.dm

```diff
@@ -590,26 +590,25 @@ var/list/rank_prefix = list(\
 ///eyecheck()
 ///Returns a number between -1 to 2
 /mob/living/carbon/human/eyecheck()
-	if(!species.has_organ[BP_EYES]) //No eyes, can't hurt them.
+	if(!species.has_process[OP_EYES]) //No eyes, can't hurt them.
 		return FLASH_PROTECTION_MAJOR
 
-	if(internal_organs_by_name[BP_EYES]) // Eyes are fucked, not a 'weak point'.
-		var/obj/item/organ/I = internal_organs_by_name[BP_EYES]
-		if(I.status & ORGAN_CUT_AWAY)
-			return FLASH_PROTECTION_MAJOR
+	var/eye_efficiency = get_organ_efficiency(OP_EYES)
+	if(eye_efficiency == 0)
```

Comment:
Actually this is fine but I would like it to cover a greater variety of situations, for the future mostly. all code that says "efficiency == 0" should be replaced by efficiency <= 0.

Reply:
Fixed.